app/controllers/concerns/shopify_app/shop_access_scopes_verification.rb

```diff
@@ -33,8 +29,14 @@ def current_shopify_domain
       ShopifyApp::Utils.sanitize_shop_domain(params[:shop])
     end
 
-    def shop_login
-      ShopifyApp::Utils.shop_login_url(shop: params[:shop], host: params[:host], return_to: request.fullpath)
+    def reauthorized_shop_login_url
+      ShopifyApp::Utils.shop_login_url(shop: params[:shop], host: params[:host], return_to: request.fullpath,
+        reauthorize: true)
+    end
+
+    def fullpage_redirect_to(url)
+      render("shopify_app/shared/redirect", layout: false,
```

Comment:
This is very similar to the method in `LoginProtection` but I didn't think it was worth extracting out.